### ISSUE TITLE
[defect]: cross compiling failed freeradius4

### ISSUE THREAD

**Issue by esaaprillia** [FreeRADIUS/freeradius-server#5399]
### What type of defect/bug is this?

Non compliance with a standards document

### How can the issue be reproduced?

i tried to cross compile freeradius-server-4.0~alpha openwrt but i got an error.

```
define Build/Compile
	# Compile jlibtool for the host once the configuration is done
	(cd $(PKG_BUILD_DIR); \
		mkdir -p build/make; \
		$(HOSTCC) $(HOST_CFLAGS) scripts/jlibtool.c -o build/make/jlibtool; \
	)
	$(MAKE) -C $(PKG_BUILD_DIR) \
		R="$(PKG_INSTALL_DIR)" \
		INSTALLSTRIP="" \
		all install
endef
```

```
2024-08-10T09:48:54.0399988Z touch /home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius-server-4.0~alpha1/.configured_ecbf4d574966879a0e79695f7f2cc341
2024-08-10T09:48:54.0426927Z rm -f /home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius-server-4.0~alpha1/.built
2024-08-10T09:48:54.0457443Z touch /home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius-server-4.0~alpha1/.built_check
2024-08-10T09:48:54.0486087Z # Compile jlibtool for the host once the configuration is done
2024-08-10T09:48:54.0508472Z (cd /home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius-server-4.0~alpha1; mkdir -p build/make; gcc -O2 -I/home/runner/work/mosdns/mosdns/sdk/staging_dir/host/include -I/home/runner/work/mosdns/mosdns/sdk/staging_dir/hostpkg/include -I/home/runner/work/mosdns/mosdns/sdk/staging_dir/target-aarch64_cortex-a53_musl/host/include scripts/jlibtool.c -o build/make/jlibtool; )
2024-08-10T09:48:54.0850529Z scripts/jlibtool.c: In function 'explode_static_lib':
2024-08-10T09:48:54.0851675Z scripts/jlibtool.c:1923:9: warning: ignoring return value of 'getcwd' declared with attribute 'warn_unused_result' [-Wunused-result]
2024-08-10T09:48:54.0852465Z  1923 |         getcwd(savewd, sizeof(savewd));
2024-08-10T09:48:54.0852834Z       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2024-08-10T09:48:54.0853659Z scripts/jlibtool.c:1948:9: warning: ignoring return value of 'chdir' declared with attribute 'warn_unused_result' [-Wunused-result]
2024-08-10T09:48:54.0854403Z  1948 |         chdir(savewd);
2024-08-10T09:48:54.0855159Z       |         ^~~~~~~~~~~~~
2024-08-10T09:48:54.4833640Z make -C /home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius-server-4.0~alpha1 R="/home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius-server-4.0~alpha1/ipkg-install" INSTALLSTRIP="" all install
2024-08-10T09:48:54.4865001Z make[3]: Entering directory '/home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius-server-4.0~alpha1'
2024-08-10T09:49:00.3738180Z Makefile:168: build/lib/.libs/libfreeradius-make-dlopen.so: cannot open shared object file: No such file or directory
2024-08-10T09:49:00.3740203Z Makefile:168: *** build/lib/.libs/libfreeradius-make-dlopen.so: failed to load.  Stop.
2024-08-10T09:49:00.3782687Z gmake[3]: Entering directory '/home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius-server-4.0~alpha1' LN-SF src/include src/freeradius-devel HEADER src/include/features.h HEADER src/include/missing.h HEADER protocol/arp/rfc826.h HEADER protocol/bfd/rfc5880.h HEADER protocol/dhcpv4/rfc2131.h HEADER protocol/dhcpv4/rfc2241.h HEADER protocol/dhcpv4/rfc2242.h HEADER protocol/dhcpv4/rfc2485.h HEADER protocol/dhcpv4/rfc2563.h HEADER protocol/dhcpv4/rfc2610.h HEADER protocol/dhcpv4/rfc2937.h HEADER protocol/dhcpv4/rfc3004.h HEADER protocol/dhcpv4/rfc3011.h HEADER protocol/dhcpv4/rfc3046.h HEADER protocol/dhcpv4/rfc3118.h HEADER protocol/dhcpv4/rfc3361.h HEADER protocol/dhcpv4/rfc3397.h HEADER protocol/dhcpv4/rfc3442.h HEADER protocol/dhcpv4/rfc3495.h HEADER protocol/dhcpv4/rfc3679.h HEADER protocol/dhcpv4/rfc3825.h HEADER protocol/dhcpv4/rfc3925.h HEADER protocol/dhcpv4/rfc4039.h HEADER protocol/dhcpv4/rfc4174.h HEADER protocol/dhcpv4/rfc4280.h HEADER protocol/dhcpv4/rfc4388.h HEADER protocol/dhcpv4/rfc4578.h HEADER protocol/dhcpv4/rfc4702.h HEADER protocol/dhcpv4/rfc4776.h HEADER protocol/dhcpv4/rfc4833.h HEADER protocol/dhcpv4/rfc5071.h HEADER protocol/dhcpv4/rfc5192.h HEADER protocol/dhcpv4/rfc5223.h HEADER protocol/dhcpv4/rfc5417.h HEADER protocol/dhcpv4/rfc5678.h HEADER protocol/dhcpv4/rfc5859.h HEADER protocol/dhcpv4/rfc5969.h HEADER protocol/dhcpv4/rfc5986.h HEADER protocol/dhcpv4/rfc6011.h HEADER protocol/dhcpv4/rfc6153.h HEADER protocol/dhcpv4/rfc6656.h HEADER protocol/dhcpv4/rfc6926.h HEADER protocol/dhcpv4/rfc7291.h HEADER protocol/dhcpv4/rfc7710.h HEADER protocol/dhcpv4/rfc7839.h HEADER protocol/dhcpv4/rfc8357.h HEADER protocol/dhcpv4/rfc8925.h HEADER protocol/dhcpv6/rfc3315.h HEADER protocol/dhcpv6/rfc3319.h HEADER protocol/dhcpv6/rfc3633.h HEADER protocol/dhcpv6/rfc3646.h HEADER protocol/dhcpv6/rfc3898.h HEADER protocol/dhcpv6/rfc4075.h HEADER protocol/dhcpv6/rfc4242.h HEADER protocol/dhcpv6/rfc4280.h HEADER protocol/dhcpv6/rfc4580.h HEADER protocol/dhcpv6/rfc4649.h HEADER protocol/dhcpv6/rfc4704.h HEADER protocol/dhcpv6/rfc4776.h HEADER protocol/dhcpv6/rfc4833.h HEADER protocol/dhcpv6/rfc4994.h HEADER protocol/dhcpv6/rfc5007.h HEADER protocol/dhcpv6/rfc5192.h HEADER protocol/dhcpv6/rfc5223.h HEADER protocol/dhcpv6/rfc5417.h HEADER protocol/dhcpv6/rfc5460.h HEADER protocol/dhcpv6/rfc5678.h HEADER protocol/dhcpv6/rfc5908.h HEADER protocol/dhcpv6/rfc5970.h HEADER protocol/dhcpv6/rfc5986.h HEADER protocol/dhcpv6/rfc6011.h HEADER protocol/dhcpv6/rfc6153.h HEADER protocol/dhcpv6/rfc6225.h HEADER protocol/dhcpv6/rfc6334.h HEADER protocol/dhcpv6/rfc6355.h HEADER protocol/dhcpv6/rfc6422.h HEADER protocol/dhcpv6/rfc6440.h HEADER protocol/dhcpv6/rfc6603.h HEADER protocol/dhcpv6/rfc6607.h HEADER protocol/dhcpv6/rfc6610.h HEADER protocol/dhcpv6/rfc6731.h HEADER protocol/dhcpv6/rfc6784.h HEADER protocol/dhcpv6/rfc6939.h HEADER protocol/dhcpv6/rfc6977.h HEADER protocol/dhcpv6/rfc7037.h HEADER protocol/dhcpv6/rfc7078.h HEADER protocol/dhcpv6/rfc7083.h HEADER protocol/dhcpv6/rfc7291.h HEADER protocol/dhcpv6/rfc7341.h HEADER protocol/dhcpv6/rfc7598.h HEADER protocol/dhcpv6/rfc7600.h HEADER protocol/dhcpv6/rfc7653.h HEADER protocol/dhcpv6/rfc7710.h HEADER protocol/dhcpv6/rfc7774.h HEADER protocol/dhcpv6/rfc7839.h HEADER protocol/dhcpv6/rfc8026.h HEADER protocol/dhcpv6/rfc8115.h HEADER protocol/dhcpv6/rfc8156.h HEADER protocol/dns/rfc1034.h HEADER protocol/eap/aka-sim/rfc4187.h HEADER protocol/eap/aka-sim/rfc5448.h HEADER protocol/eap/aka-sim/rfc7458.h HEADER protocol/radius/rfc2865.h HEADER protocol/radius/rfc2866.h HEADER protocol/radius/rfc2867.h HEADER protocol/radius/rfc2868.h HEADER protocol/radius/rfc2869.h HEADER protocol/radius/rfc3162.h HEADER protocol/radius/rfc3576.h HEADER protocol/radius/rfc3580.h HEADER protocol/radius/rfc4072.h HEADER protocol/radius/rfc4372.h HEADER protocol/radius/rfc4603.h HEADER protocol/radius/rfc4675.h HEADER protocol/radius/rfc4679.h HEADER protocol/radius/rfc4818.h HEADER protocol/radius/rfc4849.h HEADER protocol/radius/rfc5090.h HEADER protocol/radius/rfc5176.h HEADER protocol/radius/rfc5447.h HEADER protocol/radius/rfc5580.h HEADER protocol/radius/rfc5607.h HEADER protocol/radius/rfc5904.h HEADER protocol/radius/rfc6519.h HEADER protocol/radius/rfc6572.h HEADER protocol/radius/rfc6677.h HEADER protocol/radius/rfc6911.h HEADER protocol/radius/rfc6929.h HEADER protocol/radius/rfc6930.h HEADER protocol/radius/rfc7055.h HEADER protocol/radius/rfc7155.h HEADER protocol/radius/rfc7268.h HEADER protocol/radius/rfc7499.h HEADER protocol/radius/rfc7930.h HEADER protocol/radius/rfc8045.h HEADER protocol/radius/rfc8559.h HEADER protocol/radius/rfc9445.h HEADER protocol/tacacs/rfc8907.h HEADER protocol/tftp/rfc1350.h HEADER protocol/arp/freeradius.internal.h HEADER protocol/bfd/freeradius.internal.h HEADER protocol/dhcpv4/freeradius.internal.h HEADER protocol/dhcpv6/freeradius.h HEADER protocol/dhcpv6/freeradius.internal.h HEADER protocol/dns/freeradius.internal.h HEADER protocol/eap/aka-sim/freeradius.h HEADER protocol/freeradius/freeradius.internal.h HEADER protocol/freeradius/freeradius.internal.eap.h HEADER protocol/freeradius/freeradius.internal.ippool.h HEADER protocol/freeradius/freeradius.internal.password.h HEADER protocol/freeradius/freeradius.internal.sim.h HEADER protocol/ldap/freeradius.internal.h HEADER protocol/radius/freeradius.h HEADER protocol/radius/freeradius.evs5.h HEADER protocol/radius/freeradius.internal.h HEADER protocol/snmp/freeradius.h HEADER protocol/tacacs/freeradius.internal.h HEADER protocol/tftp/freeradius.internal.h HEADER protocol/tls/freeradius.h HEADER protocol/arp/dictionary.h HEADER protocol/bfd/dictionary.h HEADER protocol/dhcpv4/dictionary.h HEADER protocol/dhcpv6/dictionary.h HEADER protocol/dns/dictionary.h HEADER protocol/eap/aka-sim/dictionary.h HEADER protocol/ethernet/dictionary.h HEADER protocol/freeradius/dictionary.h HEADER protocol/ldap/dictionary.h HEADER protocol/radius/dictionary.h HEADER protocol/snmp/dictionary.h HEADER protocol/tacacs/dictionary.h HEADER protocol/vmps/dictionary.h HEADER protocol/tftp/dictionary.h HEADER protocol/tls/dictionary.h HEADER protocol/vmps/vmps.h HEADER protocol/tacacs/tacacs.h HEADER protocol/base.h CC scripts/build/make/dlopen.c CC scripts/build/make/log.c LINK build/lib/libfreeradius-make-dlopen.la CC scripts/build/make/version.c LINK build/lib/libfreeradius-make-version.la CC scripts/build/make/util.c LINK build/lib/libfreeradius-make-util.la gmake[3]: Leaving directory '/home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius-server-4.0~alpha1'
2024-08-10T09:49:00.3812311Z make[2]: *** [Makefile:1093: /home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius-server-4.0~alpha1/.built] Error 2
2024-08-10T09:49:00.3813307Z make[1]: *** [package/Makefile:116: package/feeds/mosdns/freeradius4/compile] Error 1
2024-08-10T09:49:00.3814226Z make: *** [/home/runner/work/mosdns/mosdns/sdk/include/toplevel.mk:223: package/luci-app-mosdns/compile] Error 2
2024-08-10T09:49:00.3815443Z make[3]: Leaving directory '/home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius-server-4.0~alpha1'
2024-08-10T09:49:00.3816467Z make[2]: Leaving directory '/home/runner/work/mosdns/mosdns/sdk/feeds/mosdns/freeradius4'
2024-08-10T09:49:00.3817121Z time: package/feeds/mosdns/freeradius4/compile#39.53#27.60#58.18
2024-08-10T09:49:00.3817756Z     ERROR: package/feeds/mosdns/freeradius4 failed to build.
2024-08-10T09:49:00.3818485Z make[1]: Leaving directory '/home/runner/work/mosdns/mosdns/sdk'
2024-08-10T09:49:00.3830663Z ##[error]Process completed with exit code 2.
```

### Log output from the FreeRADIUS daemon

```shell
none
```

**Comment by alandekok** [comment]
We're happy to accept patches to fix this.

Cross-compiling is not officially supported.  As such, if it works, great.  If not, then it is a very low priority to fix.

**Comment by esaaprillia** [comment]
> We're happy to accept patches to fix this.
> 
> Cross-compiling is not officially supported. As such, if it works, great. If not, then it is a very low priority to fix.

@alandekok 

i tested on master branch but got another error

```
2024-08-12T09:16:00.6003583Z CC src/lib/unlang/base.c
2024-08-12T09:16:00.7276648Z CC src/lib/unlang/call.c
2024-08-12T09:16:00.8637350Z CC src/lib/unlang/call_env.c
2024-08-12T09:16:01.0862183Z CC src/lib/unlang/caller.c
2024-08-12T09:16:01.2020224Z CC src/lib/unlang/catch.c
2024-08-12T09:16:01.3252999Z CC src/lib/unlang/compile.c
2024-08-12T09:16:02.2192157Z CC src/lib/unlang/condition.c
2024-08-12T09:16:02.3444834Z CC src/lib/unlang/detach.c
2024-08-12T09:16:02.4559135Z CC src/lib/unlang/edit.c
2024-08-12T09:16:02.8271512Z CC src/lib/unlang/foreach.c
2024-08-12T09:16:02.9943449Z CC src/lib/unlang/function.c
2024-08-12T09:16:03.1356606Z CC src/lib/unlang/group.c
2024-08-12T09:16:03.2504709Z CC src/lib/unlang/interpret.c
2024-08-12T09:16:03.6790389Z CC src/lib/unlang/interpret_synchronous.c
2024-08-12T09:16:03.8298733Z CC src/lib/unlang/io.c
2024-08-12T09:16:03.9415922Z CC src/lib/unlang/limit.c
2024-08-12T09:16:04.0654255Z CC src/lib/unlang/load_balance.c
2024-08-12T09:16:04.2027530Z CC src/lib/unlang/map.c
2024-08-12T09:16:04.3857359Z CC src/lib/unlang/mod_action.c
2024-08-12T09:16:04.4484259Z CC src/lib/unlang/module.c
2024-08-12T09:16:04.6680395Z CC src/lib/unlang/parallel.c
2024-08-12T09:16:04.8363934Z CC src/lib/unlang/return.c
2024-08-12T09:16:04.9493475Z CC src/lib/unlang/subrequest.c
2024-08-12T09:16:05.0988148Z CC src/lib/unlang/subrequest_child.c
2024-08-12T09:16:05.2562621Z CC src/lib/unlang/switch.c
2024-08-12T09:16:05.3827492Z CC src/lib/unlang/timeout.c
2024-08-12T09:16:05.5175967Z CC src/lib/unlang/tmpl.c
2024-08-12T09:16:05.6884608Z CC src/lib/unlang/try.c
2024-08-12T09:16:05.8001037Z CC src/lib/unlang/transaction.c
2024-08-12T09:16:05.9244213Z CC src/lib/unlang/xlat.c
2024-08-12T09:16:06.1129262Z CC src/lib/unlang/xlat_alloc.c
2024-08-12T09:16:06.2790003Z CC src/lib/unlang/xlat_builtin.c
2024-08-12T09:16:06.4053409Z src/lib/unlang/xlat_builtin.c: In function 'xlat_func_blake2s_256':
2024-08-12T09:16:06.4055608Z src/lib/unlang/xlat_builtin.c:2893:50: warning: implicit declaration of function 'EVP_blake2s256'; did you mean 'LN_blake2s256'? [-Wimplicit-function-declaration]
2024-08-12T09:16:06.4057327Z   return xlat_evp_md(ctx, out, xctx, request, in, EVP_##_md_func());\
2024-08-12T09:16:06.4058113Z                                                   ^~~~
2024-08-12T09:16:06.4059122Z src/lib/unlang/xlat_builtin.c:2901:1: note: in expansion of macro 'EVP_MD_XLAT'
2024-08-12T09:16:06.4059981Z  EVP_MD_XLAT(blake2s_256, blake2s256)
2024-08-12T09:16:06.4060476Z  ^~~~~~~~~~~
2024-08-12T09:16:06.4061814Z src/lib/unlang/xlat_builtin.c:2893:50: warning: passing argument 6 of 'xlat_evp_md' makes pointer from integer without a cast [-Wint-conversion]
2024-08-12T09:16:06.4063327Z   return xlat_evp_md(ctx, out, xctx, request, in, EVP_##_md_func());\
2024-08-12T09:16:06.4064132Z                                                   ^~~~~~~~~~~~~~~~
2024-08-12T09:16:06.4065146Z src/lib/unlang/xlat_builtin.c:2901:1: note: in expansion of macro 'EVP_MD_XLAT'
2024-08-12T09:16:06.4066295Z  EVP_MD_XLAT(blake2s_256, blake2s256)
2024-08-12T09:16:06.4066782Z  ^~~~~~~~~~~
2024-08-12T09:16:06.4067977Z src/lib/unlang/xlat_builtin.c:2859:74: note: expected 'const EVP_MD *' {aka 'const struct evp_md_st *'} but argument is of type 'int'
2024-08-12T09:16:06.4069413Z       UNUSED request_t *request, fr_value_box_list_t *args, EVP_MD const *md)
2024-08-12T09:16:06.4070243Z                                                             ~~~~~~~~~~~~~~^~
2024-08-12T09:16:06.4087700Z src/lib/unlang/xlat_builtin.c: In function 'xlat_func_blake2b_512':
2024-08-12T09:16:06.4089536Z src/lib/unlang/xlat_builtin.c:2893:50: warning: implicit declaration of function 'EVP_blake2b512'; did you mean 'NID_blake2b512'? [-Wimplicit-function-declaration]
2024-08-12T09:16:06.4091223Z   return xlat_evp_md(ctx, out, xctx, request, in, EVP_##_md_func());\
2024-08-12T09:16:06.4091824Z                                                   ^~~~
2024-08-12T09:16:06.4092586Z src/lib/unlang/xlat_builtin.c:2902:1: note: in expansion of macro 'EVP_MD_XLAT'
2024-08-12T09:16:06.4093253Z  EVP_MD_XLAT(blake2b_512, blake2b512)
2024-08-12T09:16:06.4093578Z  ^~~~~~~~~~~
2024-08-12T09:16:06.4094401Z src/lib/unlang/xlat_builtin.c:2893:50: warning: passing argument 6 of 'xlat_evp_md' makes pointer from integer without a cast [-Wint-conversion]
2024-08-12T09:16:06.4095309Z   return xlat_evp_md(ctx, out, xctx, request, in, EVP_##_md_func());\
2024-08-12T09:16:06.4095801Z                                                   ^~~~~~~~~~~~~~~~
2024-08-12T09:16:06.4096428Z src/lib/unlang/xlat_builtin.c:2902:1: note: in expansion of macro 'EVP_MD_XLAT'
2024-08-12T09:16:06.4096957Z  EVP_MD_XLAT(blake2b_512, blake2b512)
2024-08-12T09:16:06.4097272Z  ^~~~~~~~~~~
2024-08-12T09:16:06.4098022Z src/lib/unlang/xlat_builtin.c:2859:74: note: expected 'const EVP_MD *' {aka 'const struct evp_md_st *'} but argument is of type 'int'
2024-08-12T09:16:06.4098895Z       UNUSED request_t *request, fr_value_box_list_t *args, EVP_MD const *md)
2024-08-12T09:16:06.4099416Z                                                             ~~~~~~~~~~~~~~^~
2024-08-12T09:16:07.2545133Z CC src/lib/unlang/xlat_eval.c
2024-08-12T09:16:07.7347118Z CC src/lib/unlang/xlat_expr.c
2024-08-12T09:16:08.4379199Z CC src/lib/unlang/xlat_func.c
2024-08-12T09:16:08.6138892Z CC src/lib/unlang/xlat_inst.c
2024-08-12T09:16:08.8204634Z CC src/lib/unlang/xlat_pair.c
2024-08-12T09:16:08.9420020Z CC src/lib/unlang/xlat_purify.c
2024-08-12T09:16:09.1279459Z CC src/lib/unlang/xlat_redundant.c
2024-08-12T09:16:09.2858884Z src/lib/unlang/xlat_redundant.c: In function 'xlat_register_redundant':
2024-08-12T09:16:09.2860680Z src/lib/unlang/xlat_redundant.c:619:4: warning: 'flags' may be used uninitialized in this function [-Wmaybe-uninitialized]
2024-08-12T09:16:09.2866156Z     xlat_func_flags_set(xlat, flags);
2024-08-12T09:16:09.2866727Z     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2024-08-12T09:16:09.3442357Z CC src/lib/unlang/xlat_tokenize.c
2024-08-12T09:16:09.9199453Z CC src/lib/sim/comp128.c
2024-08-12T09:16:10.0148763Z CC src/lib/sim/milenage.c
2024-08-12T09:16:10.2442058Z CC src/lib/sim/ts_34_108.c
2024-08-12T09:16:10.5644808Z LINK /home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius4-4.0/build/lib/libfreeradius-server.la
2024-08-12T09:16:10.7531704Z LINK /home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius4-4.0/build/lib/libfreeradius-unlang.la
2024-08-12T09:16:10.9152401Z LINK /home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius4-4.0/build/lib/libfreeradius-sim.la
2024-08-12T09:16:11.0257276Z LINK /home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius4-4.0/build/lib/libfreeradius-eap-aka-sim.la
2024-08-12T09:16:11.1533851Z CC src/lib/server/pair_server_tests.c
2024-08-12T09:16:11.5929990Z /home/runner/work/mosdns/mosdns/sdk/staging_dir/toolchain-aarch64_cortex-a53_gcc-8.4.0_musl/bin/../lib/gcc/aarch64-openwrt-linux-musl/8.4.0/../../../../aarch64-openwrt-linux-musl/bin/ld: /home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius4-4.0/build/lib/local/.libs/libfreeradius-unlang.so: undefined reference to `EVP_blake2b512'
2024-08-12T09:16:11.5934828Z /home/runner/work/mosdns/mosdns/sdk/staging_dir/toolchain-aarch64_cortex-a53_gcc-8.4.0_musl/bin/../lib/gcc/aarch64-openwrt-linux-musl/8.4.0/../../../../aarch64-openwrt-linux-musl/bin/ld: /home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius4-4.0/build/lib/local/.libs/libfreeradius-unlang.so: undefined reference to `EVP_blake2s256'
2024-08-12T09:16:11.5937618Z collect2: error: ld returned 1 exit status
2024-08-12T09:16:11.5939377Z make[3]: *** [scripts/boiler.mk:731: /home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius4-4.0/build/bin/local/pair_server_tests] Error 1
2024-08-12T09:16:11.5941626Z make[3]: Leaving directory '/home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius4-4.0'
2024-08-12T09:16:11.5967043Z make[2]: Leaving directory '/home/runner/work/mosdns/mosdns/sdk/feeds/mosdns/freeradius4'
2024-08-12T09:16:11.5968874Z make[2]: *** [Makefile:1113: /home/runner/work/mosdns/mosdns/sdk/build_dir/target-aarch64_cortex-a53_musl/freeradius4-4.0/.built] Error 2
2024-08-12T09:16:11.5971633Z time: package/feeds/mosdns/freeradius4/compile#138.55#47.30#159.51
```

**Comment by arr2036** [comment]
Cross compiling is mostly supported in v4 from some work I did with WASM a while back.  I've added autoconf checks for the blake functions.  Let me know if there are any other issues.

**Comment by esaaprillia** [comment]
> Cross compiling is mostly supported in v4 from some work I did with WASM a while back. I've added autoconf checks for the blake functions. Let me know if there are any other issues.

I'll update you here if I get any other errors

**Comment by esaaprillia** [comment]
hy @arr2036 

maybe you are interested in the problems faced by freeradius-3.2.5 in openwrt. I hope you see it

https://downloads.openwrt.org/snapshots/faillogs/aarch64_generic/packages/freeradius3/compile.txt

**Comment by arr2036** [comment]
v3 has no guarantees of cross-compilation compatibility, only v4.  It was a fair amount of work to get the build system working correctly, and there are higher priorities than back porting that to v3.  As Alan says, patches welcome :)

**Comment by esaaprillia** [comment]
> v3 has no guarantees of cross-compilation compatibility, only v4. It was a fair amount of work to get the build system working correctly, and there are higher priorities than back porting that to v3. As Alan says, patches welcome :)

fix

ac_cv_header_readline_h=no
ac_cv_header_readline_readline_h=no
ac_cv_header_readline_history_h=no

**Comment by arr2036** [comment]
Those are variables set by the autoconf script... we can't change their values.

**Comment by esaaprillia** [comment]
> Those are variables set by the autoconf script... we can't change their values.

https://github.com/openwrt/packages/blob/028474f0c9baa1daa748f3d7656bd1d472ddf697/net/freeradius3/Makefile#L669
